PROBLEM SHEET 1: The aim of this first problem sheet is to introduce you to the environment in which we will be running Python during the post exam period. The computers in the teaching lab have several version of Python installed and as we will see it can be used in different ways. When we return to onsite lab sessions we will be using the Anaconda installation, with a Python 3.x version. However for now we will use the CoCalc.com cloud based servers accessed through a web browser.



# Problem Sheet 1, Ex1: Creating new cells in the Jupyter notebook

* Create a new **Code** cell below, inside the **code** cell create a variable called *weekday* and set it's value to the *string* that is the name of the day when you type it (i.e Monday)
* Create a new **Markdown** cell below, inside the **Markdown** cell write your name and candidate number in bold, make a indented list (i.e. like this list of questions, double clicking on a box shows you the code) of your three favourite films and use $\LaTeX$ math mode to write down Einstein's equation that shows that energy and mass are interchangeable.

# Problem Sheet 1, Ex2: Using Python as a calculator

In the code window below (marked enter your answer for EX2 here), calculate and print to screen

* The result of 4/7 where 4 and 7 are integer numbers
* The square root of nineteen
* The factorial of 19
* The sine of 37 degrees

In [5]:
# Using the hash symbol allows you to include comments in your code which will not be interpreted by python
# If I type a list of calculations, such as:
3+4
5*6
22/3.3
#and then run the code cell (either by holding shift+enter or the run button)


6.666666666666667

In [6]:
#Then I only see the last output, I did not tell Python to do anything with the calculations, if however I use the print() function:
print(3+4)
print(5*6)
print(22/3.3) #Also notice that the Jupyter notebooks has automatically change the colour for comments and functions etc

7
30
6.666666666666667


In [7]:
# Then I see that all the values have been printed to the screen, I can learn more about how print() works by using the help() function, with print as the argument (the value inside the brackets)
help(print) #this displays information about the function called it's docstring

Help on built-in function print in module builtins:

print(...)
    print(value, ..., sep=' ', end='\n', file=sys.stdout, flush=False)
    
    Prints the values to a stream, or to sys.stdout by default.
    Optional keyword arguments:
    file:  a file-like object (stream); defaults to the current sys.stdout.
    sep:   string inserted between values, default a space.
    end:   string appended after the last value, default a newline.
    flush: whether to forcibly flush the stream.



In [8]:
# Note that some of these basic maths operations are not available in the standard python library if you try
sin(10)
# You will recieve a NameError: name 'sin'is not defined
# The functions can be found in the math module, to access this you will first need to import the functions into your code (i.e. import math) loads the names, dir(math) will return a list of the names of all of the functions in the math module,
# math.funcname() will let you use function (i.e to find the log of 10, you type math.log(10), this tells python to use the log function from the math module)
# Use the help() function to learn what the function in the module do.

NameError: name 'sin' is not defined

In [ ]:
# Enter your answer for EX2 here:


# Problem Sheet 1, EX3: Requesting inputs from users and returning the output after evaluation:

* A ball is dropped from a tower of height $h$, built on level ground. The ball has initial velocity zero and accelerates downwards under gravity.
* Write a program that asks the user to enter the height in meters of the tower and a time interval $t$ in seconds (using the input function), then prints on the screen the height of the ball from the ground at time $t$ after it is dropped, ignoring air resistance.

In [ ]:
# Enter your answer for EX3 here:


# Problem Sheet 1, Ex4: A question to practice requesting inputs from users and returning the output after evaluation:


A satellite is to be launched into a circular orbit around the Earth so that it orbits the planet once every $T$ seconds.

*  Show (in the Markdown cell) that the altitude $h$ above the Earths surface that the satellite must have is $h = (GMT^2/4\pi^2)^{1/3} -R$, where $G = 6.67$x$10^{-11}\,\mathrm{m}^3\mathrm{kg}^{-1}\mathrm{s}^{-2}$ is Newtons gravitational constant, $M = 5.97$x$10^{24}\,\mathrm{kg}$ is the mass of the Earth, and $R = 6371\,\mathrm{km}$ is its radius. Before writing a program it is important that you understand what your variables are and what you want the algorithm to do.
* Write a program that asks the user to enter the desired value of $T$ and then calculates and prints out the correct altitude in meters.
* Use your program to calculate the altitudes of satellites that orbit the Earth once a day (so-called geosynchronous orbit), once every 90 minutes, and once every 45 minutes.
* What do you conclude from the last of these calculations?



Write your answer to the first part here (**Hint:** Remind yourself of Kepler's Third Law)


In [ ]:
import math
# Enter your code for EX4 here (write a comment for the last part at the end of the code)


# Problem Sheet 1, Ex5: Writing to a file.

* Using the write() method described in the notes, create a file containing the first law of thermodynamic:
 
``In all cases in which work is produced by the agency of heat, a quantity of heat is consumed which
is proportional to the work done; and conversely, by the expenditure of an equal quantity of work an equal quantity of heat is produced.''

* Write a program to read out the sixth word of the string and then print the whole statement.

* Print (to screen) a list of the functions that can operate on the string, \emph{hint type the string name in the interpreter window followed by a '.' and then press tab (auto complete) to see available functions}

In [ ]:
# Type your code for Ex4 here, I have cut a bit from the notes to remind you:
# Open a file
fo = open("1lt.txt", "w" ) # makes an file object fo (could have assigned anyname here) which is the open file foo.txt, if it does not exist it will make it


# Problem Sheet 1, Ex6: Saving your work

* We will be marking Problem Sheet 1 in the first week of the Autumn term, and files will be submitted via turnitin on the Moodle page, you should still have access to CoCalc in the Autumn but to ensure that you have access to the work that you have done download the code and save it on your favourite storage mechanism (I suggest your university y-drive, dropbox, and usb stick) each year someone always loses their work (laptops died or lost/stolen etc)

### Next Step

Either return to the [Start Menu](PH2150_Start_here.ipynb), or move on to [Problem Sheet 2](Problem_Sheet_2.ipynb)
